databuilder/query_engines/base_sql.py

```diff
@@ -63,6 +71,9 @@ class BaseSQLQueryEngine(BaseQueryEngine):
     # Per-instance cache for SQLAlchemy Engine
     _engine: Optional[sqlalchemy.engine.Engine] = None
 
+    # Force subclasses to define this
+    temp_table_prefix: str = MissingString("'temp_table_prefix' is undefined")
```

Comment:
Nothing to do for this PR, but I'm hoping we can have mypy do this for us at some point!